library-management-system/src/main/java/com/johnny/training/Library.java

```
@@ -0,0 +1,86 @@
+package com.johnny.training;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class Library {
+    private List<Book> allBooks;
+    public static Map<Member , List<Book>> checkedOutBooks;
+
+    public Library(){
+        allBooks = new ArrayList<>();
+        checkedOutBooks = new HashMap<>();
+    }
+
+    public List<Book> getAllBooks() {
+        return allBooks;
+    }
+
+    public void setAllBooks(List<Book> allBooks) {
+        this.allBooks = allBooks;
+    }
+
+    public Map<Member, List<Book>> getCheckedOutBooks() {
+        return checkedOutBooks;
+    }
+
+    public void setCheckedOutBooks(Map<Member, List<Book>> checkedOutBooks) {
+        this.checkedOutBooks = checkedOutBooks;
+    }
+    public void addBook (Book book){
+        allBooks.add(book);
+    }
+    public List<Book> searchByTitle(String title){
+        List<Book> result = new ArrayList<>();
+        for(Book book : allBooks){
+            if(book.getTitle().equalsIgnoreCase(title)){
+                result.add(book);
+            }
+        }
+        return result;
+    }
+    public List<Book> searchByAuthor(String author){
+        List<Book> result = new ArrayList<>();
+        for(Book book : allBooks){
+            if(book.getAuthor().equalsIgnoreCase(author)){
+                result.add(book);
+            }
+        }
+        return result;
+    }
+    public Book searchByISBN(int ISBN){
```

In case the ISBN was not found, returning a null might cause a NullPointerException if the clients are using the book without checking for null.
A better approach would be to throw a dedicated exception in case the book was not found, e.g: BookNotFoundException. Consider creating this exception and throwing it here instead of returning null.
Also, consider also creating a method streamByISBN that uses the java streams to perform the same behavior